static/js/containers/SettingsPage.js

```diff
@@ -0,0 +1,100 @@
+/* global SETTINGS: false */
+import React from 'react';
+import { connect } from 'react-redux';
+
+import Jumbotron from '../components/Jumbotron';
+import PrivacyForm from '../components/PrivacyForm';
+import ProfileFormContainer from './ProfileFormContainer'
+import {
+  startProfileEdit,
+  updateProfile,
+  updateProfileValidation,
+  clearProfileEdit,
+  saveProfile,
+} from '../actions';
+
+class SettingsPage extends React.Component {
+
+  static propTypes = {
+    dispatch:       React.PropTypes.func.isRequired,
+    updateProfile:  React.PropTypes.func.isRequired,
+    saveProfile:    React.PropTypes.func.isRequired,
+    userProfile:    React.PropTypes.object.isRequired,
+    ui:             React.PropTypes.object.isRequired,
+    errors:         React.PropTypes.object,
+    isSettingPage:  React.PropTypes.bool
+  };
+
+  constructor(props) {
+    super(props);
+    this.updateProfile = this.updateProfile.bind(this);
+    this.saveProfile = this.saveProfile.bind(this);
+  }
+
+  render() {
+    const { userProfile: { profile }, ui, errors, isSettingPage } = this.props;
+    let preferredName = profile.preferredName || SETTINGS.name;
```

Comment:
we have a `getPreferredName` function in `util/util.js` which you could use here instead.